app/models/cart.rb

```diff
@@ -23,12 +23,39 @@ def items
   end
 
   def subtotal(item)
-    item.price * @contents[item.id.to_s]
+    if highest_discount(item) != nil
+      discounted(item)
+    else
+      item.price * @contents[item.id.to_s]
+    end
```

Comment:
Add conditional to check if the item has a discount. If it does it calls the method that calculates the new subtotal